source/System.cpp

```diff
@@ -757,6 +714,34 @@ void System::LoadObject(const DataNode &node, Set<Planet> &planets, int parent)
 
 
 
+// Once the star map is fully loaded or an event has changed systems
+// or links, figure out which stars are "neighbors" of this one, i.e.
+// close enough to see or to reach via jump drive.
+void System::UpdateNeighbors(const Set<System> &systems, double distance)
+{
+	set<const System *> &neighborSet = neighbors[distance];
+	
+	// Every star system that is linked to this one is automatically a neighbor,
+	// even if it is farther away than the maximum distance.
+	for(const System *system : links)
+		if(!(system->Position().Distance(position) <= distance))
```

Comment:
This if-statement is not needed; an element will only be in a set once, even if it gets inserted multiple times.

Reply:
Looks like this was a holdover from when links and neighbors were vectors. Removed the if statement.